clang/test/CXX/class/class.union/class.union.anon/p4.cpp

```diff
@@ -8,3 +8,13 @@ union U {
     int y = 1; // expected-error {{initializing multiple members of union}}
   };
 };
+
+namespace GH149985 {
```

Comment:
Can we add a test with reproducer from https://github.com/llvm/llvm-project/issues/149985 ?

Reply:
It's situational and a bit of a judgement call; it's good to use the reproducer from the issue because that is easy for reviewers to verify that it actually fixes the bug. But not all reproducers on issues will be reasonable for inclusion in the test suite, so it's fine to make modifications as needed. I think it'd be good to grab the test from the issue in addition to this; the issue is using anonymous structures within a union which means it could exercise somewhat different code paths than your test case.